client/X11/xf_window.c

```diff
@@ -942,7 +944,11 @@ void xf_SetWindowVisibilityRects(xfContext* xfc, xfAppWindow* appWindow, RECTANG
 		xrects[i].height = rects[i].bottom - rects[i].top;
 	}
 
-	XShapeCombineRectangles(xfc->display, appWindow->handle, ShapeBounding, 0, 0, xrects, nrects, ShapeSet, 0);
+	/* Translate the server rail window offset to a local offset */
+	translatedWindowOffsetX = (appWindow->visibleOffsetX - appWindow->windowOffsetX);
+	translatedWindowOffsetY = (appWindow->visibleOffsetY - appWindow->windowOffsetY);
```

Comment:
Yes, it works when windowOffsetX is negative. In fact, that's the only time I've ever witnessed translatedWindowOffsetX becoming non-zero. As I wrote in the commit message, this hunk was added to fix exactly the case when the window is partially off the left edge of the screen.